lib/engine/game/g_1856.rb

```diff
@@ -47,6 +48,96 @@ class G1856 < Base
+        return false if from.name == '470'
+        # double dits upgrade to Green cities in gray
+        return @phase.tiles.include? 'gray' if to.name == '14' && %w[55 1].include?(from.name)
+        return @phase.tiles.include? 'gray' if to.name == '15' && %w[56 2].include?(from.name)
+
+        # yellow dits upgrade to yellow cities in gray
+        return @phase.tiles.include? 'gray' if to.name == '5' && from.name == '3'
+        return @phase.tiles.include? 'gray' if to.name == '57' && from.name == '4'
+        return @phase.tiles.include? 'gray' if to.name == '6' && from.name == '58'
+
+        # yellow dits upgrade to plain track in gray
+        return @phase.tiles.include? 'gray' if to.name == '7' && from.name == '3'
+        return @phase.tiles.include? 'gray' if to.name == '9' && from.name == '4'
+        return @phase.tiles.include? 'gray' if to.name == '8' && from.name == '58'
+
+        # Certain green cities upgrade to other labels
+        return to.name == '127' if from.color == :green && from.hex.name == BARRIE_HEX
+        return to.name == '126' if from.color == :green && from.hex.name == LONDON_HEX
+        # You may lay the brown 5-spoke L if and only if it is laid on a L hex -
+        # NOT EVEN IF YOU GREEN A DOUBLE DIT ON A LAKE EDTGE
+        return to.name == '125' if from.color == :green && LAKE_HEXES.include?(from.hex.name)
+        # The L hexes on the map start as plain yellow cities
+        return %w[5 6 57].include?(to.name) if LAKE_HEXES.include?(from.hex.name) && from.color == 'white'
+        # B,L to B-L
+        return to.name == '121' if from.color == :yellow && [BARRIE_HEX, LONDON_HEX].include?(from.hex.name)
+        # Hamilton OO upgrade is yet another case of ignoring labels in upgrades
+        return to.name == '123' if from.color == :brown && from.hex.name == HAMILTON_HEX
+
+        super
+      end
+
+      #
+      # Get all possible upgrades for a tile
+      # tile: The tile to be upgraded
+      # tile_manifest: true/false Is this being called from the tile manifest screen
+      #
+      def all_potential_upgrades(tile, tile_manifest: false)
+        upgrades = super
+        return upgrades unless tile_manifest
+
+        # In phase 6+ single dits may be turned into plain yellow track or yellow cities
+        if @phase.tiles.include?('gray')
+          upgrades |= [@straight_city, @straight_track] if tile.name == '4'
+          upgrades |= [@gentle_city, @gentle_track] if tile.name == '58'
+          upgrades |= [@sharp_city, @sharp_track] if tile.name == '3'
+          # furthermore, double dits may be upgraded to green cities, if track can be preserved
+          upgrades |= [@x_city] if tile.name == '55'
+          upgrades |= [@x_city] if tile.name == '1'
+          upgrades |= [@k_city] if tile.name == '56'
+          upgrades |= [@k_city] if tile.name == '2'
+        end
+        upgrades |= @tiles.find { |t| t.name == '127' } if tile.name == '125'
```

Comment:
a 125 can upgrade to a 127??

Reply:
No; I went and fixed the tile manifest (it also shows OO -> Hamilton now)